daemon/src/taker_cfd.rs

```diff
@@ -116,10 +118,16 @@ impl<O, W> Actor<O, W> {
     }
 
     async fn handle_propose_settlement(&mut self, msg: ProposeSettlement) -> Result<()> {
-        let ProposeSettlement {
-            order_id,
-            current_price,
-        } = msg;
+        let ProposeSettlement { order_id, bid, ask } = msg;
+
+        let mut conn = self.db.acquire().await?;
+
+        #[allow(deprecated)]
+        // We need the position and role to decide the price here. Consider re-modelling access to
+        // quote / where to taker the decision to avoid this call.
+        let cfd = crate::command::load_cfd(order_id, &mut conn).await?;
```

Comment:
😞 Another chance for collaborative settlement to access the DB directly. At least it seems like it is the only protocol which doesn't fit into the mold.

Reply:
I had the same thought. Might be worth exploring some redesign choices to overcome some of these issues. For now I decided to just go forward and not think about refactoring too much 😬